File: p1.py
#Time: O(n)
#Space: O(1)
class Solution(object):
    def isRobotBounded(self, instructions):
        """
        :type instructions: str
        :rtype: bool
        """
        
        dirs = [[0,1], [1,0], [0,-1], [-1,0]] 

        i = 0
        x, y = 0,0

        for j in range(len(instructions)):
            temp = instructions[j]

            if temp == 'R':
                i = (i + 1) % 4
            elif temp == 'L':
                i = (i+3) % 4
            else:
                x += dirs[i][0]
                y += dirs[i][1]

        return (x==0 and y==0) or i > 0

File: p1.java
// Time Complexity :O(n)
// Space Complexity :O(1)
// Did this code successfully run on Leetcode :yes
// Any problem you faced while coding this :no


// Your code here along with comments explaining your approach

class Solution {
    int[][] dirs = new int[][]{{0,1} , {1,0}, {0,-1} ,{-1,0}};
    public boolean isRobotBounded(String instructions) {
        int x = 0;
        int y = 0;
        int idx = 0;
        //Run the insructions once, if we reach at the starting point, return zero
        //If we are still facing north return false
        for(int i=0; i<instructions.length(); i++){
            char c = instructions.charAt(i);
            if(c == 'G'){
                x = x + dirs[idx][0];
                y = y + dirs[idx][1];
            } else if(c == 'L'){
                idx = (idx + 3)%4;
            } else if(c == 'R'){
                idx = (idx+1)%4;
            }
        }
        if(x ==0 && y ==0) return true;
        return idx!=0;
    }
}